build/xtask/src/dist.rs

```diff
@@ -1735,7 +1762,11 @@ pub struct Allocations {
     /// Map from memory-name to address-range
     pub kernel: BTreeMap<String, Range<u32>>,
     /// Map from task-name to memory-name to address-range
-    pub tasks: BTreeMap<String, BTreeMap<String, Range<u32>>>,
+    ///
+    /// A task may have multiple address ranges in the same memory space for
+    /// efficient packing; if this is the case, the addresses will be contiguous
+    /// and each individual range will respect MPU requirements.
+    pub tasks: BTreeMap<String, BTreeMap<String, Vec<Range<u32>>>>,
```

Comment:
(I think this has an implicit assumption about vec order as well, it should go in the comment. Or maybe a BTreeSet would be better than a Vec?)

Reply:
Yup, I added a `ContinguousRanges` type which enforces that the ranges are contiguous (3d128cbee0fd75fab17799a754281d836d4a7321)